src/globalFlags.ts

```diff
@@ -0,0 +1,29 @@
+import { WorkspaceConfiguration } from "vscode";
+import { getHeaderArgs } from "./headers";
+import { escapeCommandArg } from "./util";
+
+export function getGlobalFlags(
+	configs: WorkspaceConfiguration,
+	configDir?: string,
+): string[] {
+	const globalFlags = configs.get<string[]>("coder.globalFlags") || [];
+	const headerArgs = getHeaderArgs(configs);
+	const globalConfigArgs = configDir
+		? ["--global-config", escapeCommandArg(configDir)]
+		: [];
+
+	// Precedence of "coder.headerCommand" is higher than "coder.globalConfig" with the "--header-command" flag
```

Comment:
Actually, since `coder` already has precedence in the order of arguments, we could just do something like this I think?  (or spread instead of concat)

```javascript
return (configs.get<string[]>("coder.globalFlags") || [])
  .concat(["--global-config", escapeCommandArg(configDir)]
  .concat(getHeaderArgs)
```

And the stuff at the end would automatically take precedence by the cli.